samples/aspnetapp/Dockerfile.self-contained

```
@@ -0,0 +1,23 @@
+# Learn about building .NET container images:
+# https://github.com/dotnet/dotnet-docker/blob/main/samples/README.md
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:9.0 AS build
+ARG TARGETARCH
+WORKDIR /source
+
+# Copy project file and restore as distinct layers
+COPY --link aspnetapp/*.csproj .
+RUN dotnet restore -r linux-$TARGETARCH
+
+# Copy source code and publish app
+COPY --link aspnetapp/. .
+RUN dotnet publish --self-contained -r linux-$TARGETARCH --no-restore -o /app
```

It still works like this, doesn't it? Otherwise, can we use `--use-current-runtime` instead?

```suggestion
RUN dotnet publish --self-contained -a $TARGETARCH --no-restore -o /app
```

---

Oh, right. So using `-a` doesn't work with --self-contained, unless you enable `PublishReadyToRun`. This is https://github.com/dotnet/sdk/issues/36334.

This is why I put the commented-out properties in the aspnetapp.csproj file. https://github.com/dotnet/dotnet-docker/blob/dddd8f0a53cda994b91cb50a2dfc0a9651399c12/samples/aspnetapp/aspnetapp/aspnetapp.csproj#L9-L14

If I add a comment here referring to those, it should be OK to use the `-a` option instead of `-r`.

Another option is to set the properties directly in the Dockerfile (just tested this):

```Dockerfile
# Copy project file and restore as distinct layers
COPY --link aspnetapp/*.csproj .
RUN dotnet restore -a $TARGETARCH \
    -p:PublishReadyToRun=true \
    -p:PublishSingleFile=true

# Copy source code and publish app
COPY --link aspnetapp/. .
RUN dotnet publish -a $TARGETARCH --no-restore -o /app \
    --self-contained \
    -p:PublishReadyToRun=true \
    -p:PublishSingleFile=true
```

I think if we did that, we'd need to add a comment recommending putting these in the project file instead. But that would let the Dockerfile build without modifying the project file.

---

I think this was why @richlander had removed the self-contained sample in the first place. Because we wanted to have publishing configuration defined in the project file. But that made it not easily reusable between self-contained and framework-dependent.

---

I see you re-requested review but this isn't resolved yet. Using `-r linux` isn't preferred because it wouldn't translate to Alpine without modification. Is `--use-current-runtime` an option?

---

It seems like `--ucr` works in the same scenarios as `-a $TARGETARCH` - it still needs `ReadyToRun` to be set to true. I'm OK with dropping the self-contained sample since we do still have `releasesapp` for that.

---

Yeah, I think that's probably the way to go for now.